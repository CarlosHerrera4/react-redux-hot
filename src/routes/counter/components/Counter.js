import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Fire = props =>
  <section className={`${props.className} fire-container`}>
   Fire me!
   <hr />
    <span className='fire'>ðŸ”¥</span>
    <button onClick={props.incrementCounter}>Increment</button>
    <button onClick={props.decreaseCounter}>Decrease</button>
  </section>;

const FireStyled = styled(Fire)`
  &.fire-container{
    .fire{
      font-size: ${props => `${props.counter * 16}px`};
    }
  }
`;

Fire.propTypes = {
  incrementCounter: PropTypes.func.isRequired,
  decreaseCounter: PropTypes.func.isRequired,
  className: PropTypes.string.isRequired,
};

export default FireStyled;
