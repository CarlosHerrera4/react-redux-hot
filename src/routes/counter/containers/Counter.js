import { connect } from 'react-redux';
import { decreaseCounter, incrementCounter } from '../../../modules/counter';
import Counter from '../components/Counter';

const mapDispatchToprops = {
  decreaseCounter,
  incrementCounter,
};

const mapStateToProps = state => ({
  counter: state.counter,
});

export default connect(mapStateToProps, mapDispatchToprops)(Counter);
