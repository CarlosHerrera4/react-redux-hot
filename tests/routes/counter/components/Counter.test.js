import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../../../../src/routes/counter/components/Counter';

test('Counter', () => {
  let counter = 0;
  const component = renderer.create(
    <Counter
      counter={counter}
      incrementCounter={() => { counter += 1; }}
      decreaseCounter={() => { counter -= 1; }}
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
