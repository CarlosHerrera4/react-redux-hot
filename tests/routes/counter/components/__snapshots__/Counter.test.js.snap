// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fire 1`] = `
<div>
  Hello fire route! ðŸ˜Ž
  <hr />
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    onClick={[Function]}
  >
    Decrease
  </button>
</div>
`;
